### ISSUE TITLE
 --statistics --show-source  not supported

### ISSUE THREAD

**Issue by youknowone** [charliermarsh/ruff#4140]
```console
$ ruff . --statistics --show-source
error: the argument '--statistics' cannot be used with '--show-source'
$ ruff --version
ruff 0.0.261
```
I don't have `pyproject.toml`

This is working just as expected with flake8.

I found it  while working on https://github.com/RustPython/RustPython/issues/4943

**Comment by charliermarsh** [comment]
Ah yeah, this is an intentional difference -- we treat `--statistics` a little differently, effectively as its own subcommand. When you run Ruff with `--statistics`, we _just_ print the statistics and not the violations, whereas Flake8 prints both. The thinking was that `--statistics` would be more of a one-off or few-time command vs. the standard linting workflow.

**Comment by youknowone** [comment]
Thank you for explanation!

**Comment by KotlinIsland** [comment]
@charliermarsh My ci looks like:
```yaml
 - ruff check; ruff --statistics
 ```
 
 I can't think of any reason I would not want this extra information to show.
